packages/kit/src/runtime/client/client.js

```diff
@@ -332,7 +332,12 @@ async function _invalidate() {
 	if (!navigation_result || nav_token !== token) return;
 
 	if (navigation_result.type === 'redirect') {
-		return _goto(new URL(navigation_result.location, current.url).href, {}, 1, nav_token);
+		return _goto(
+			new URL(navigation_result.location, current.url).href,
+			{ replaceState: true },
```

Comment:
I'm not sure if this is right. If the result is a redirect, we do want to create a history entry for the redirect. Can you can explain why you've made this change here?

Reply:
No, you don't want to create a history entry, because 
1) on server redirect (302) history entry is not created, and, IMO, client side and server side navigation should be consistent 
2) when you create history entry on client redirect, you create a loop in history. I mean, for example, if you press "go back" button in your browser, browser navigates to the "redirect" entry, which instantly goes forward again, so you essentially become stuck in a loop. With server redirects such loops cannot be achieved, and neither should they be achieved on the client side

Reply:
I see. So we want the intermediate navigations to not create history entries. Is that correct?